Kubernetes/HOWTO-on-prem.md

```diff
@@ -0,0 +1,103 @@
+# Kubernetes On-Premises Deployment From Scratch -- Start to Finish #
+This guide will walk you through deploying *Kubernetes 1.8* on a Linux master and join two Windows nodes to it without a cloud provider.
+
+**Note**: The guide will assume a local working directory of `~/kube` for the Kubernetes setup. If you choose a different directory, just replace any references to that path.
+
+## Building the Binaries ##
+We will begin with a Linux master node and a Windows worker node on RS3+, both of which are assumed to be VMs.
+
+First, install all of the pre-requisites on the master:
+
+    # apt-get install git build-essentials docker.io
+
+### Building Kubernetes ###
+We will need to build the `kubelet` and `kubeproxy` binaries for Windows from scratch _cross-compiling from Linux_, so let's [set up a Go environment](https://golang.org/doc/install#install) for that, first. This guide assumes Go 1.9 and has not been tested with anything else. With your Go environment ready (don't forget to set your `$GOPATH`!), we can build the Windows binaries:
+
+```bash
+$ K8SREPO="github.com/madhanrm/kubernetes"
+$ go get -d $K8SREPO

```

Comment:
Ignore error from this command, e.g. 'no Go files in $GOPATH/GitHub.com/madhanrm/kubernetes